src/fernet/core.clj

```diff
@@ -41,14 +41,17 @@
                 expected-signature))
       (invalid-token))))
 
-(defn aes128cbc [mode key iv message]
-  (.doFinal (doto (Cipher/getInstance "AES/CBC/PKCS7Padding")
-              (.init
-                (mode {:encrypt Cipher/ENCRYPT_MODE
-                       :decrypt Cipher/DECRYPT_MODE})
-                (new SecretKeySpec key "AES")
-                (new IvParameterSpec iv)))
-            message))
+(defn- aes128cbc
+  [mode key iv message]
+  (let [cipher (Cipher/getInstance "AES/CBC/PKCS7Padding")
+        mode (mode {:encrypt Cipher/ENCRYPT_MODE
+                    :decrypt Cipher/DECRYPT_MODE})
+        k (new SecretKeySpec key "AES")
+        iv (new IvParameterSpec iv)]
```

Comment:
I thought `Class.` was preferred, but the Java interop page doesn't suggest either way. that said, you use `String.` below; you should probably pick a style and stick with it.